web/pages/[username]/[contractSlug].tsx

```diff
@@ -165,64 +191,66 @@ export async function getStaticPaths() {
 }
 
 export default function ContractPage(props: {
-  contract: Contract | null
-  historyData: HistoryData
-  pointsString?: string
-  comments: ContractComment[]
-  userPositionsByOutcome: ContractMetricsByOutcome
-  totalPositions: number
-  totalBets: number
-  topContractMetrics: ContractMetric[]
-  creatorTwitter: string | undefined
-  relatedContracts: Contract[]
+  visibility: visibility
+  contractSlug: string
+  contractParams?: ContractParams
 }) {
-  props = usePropz(props, getStaticPropz) ?? {
-    contract: null,
-    historyData: { bets: [], points: [] },
-    pointsString: '',
-    comments: [],
-    userPositionsByOutcome: {},
-    totalBets: 0,
-    topContractMetrics: [],
-    totalPositions: 0,
-    relatedContracts: [],
-  }
+  const { visibility, contractSlug, contractParams } = props
+  return (
+    <Page maxWidth="max-w-[1400px]">
+      {visibility == 'private' && (
+        <PrivateContractPage contractSlug={contractSlug} />
+      )}
+      {visibility != 'private' && contractParams && (
+        <NonPrivateContractPage contractParams={contractParams} />
+      )}
+    </Page>
+  )
+}
+
+export function NonPrivateContractPage(props: {
+  contractParams: ContractParams
+}) {
+  const { contract, historyData, pointsString } = props.contractParams
 
   const inIframe = useIsIframe()
   if (inIframe) {
-    return <ContractEmbedPage {...props} />
+    return <ContractEmbedPage contract={contract} historyData={historyData} />
   }
-
-  const { contract } = props
-
   if (!contract) {
     return <Custom404 />
   }
-
-  return <ContractPageContent key={contract.id} {...{ ...props, contract }} />
+  return (
+    <>
+      <ContractSEO contract={contract} points={pointsString} />
+      <ContractPageContent
+        key={contract.id}
+        contractParams={props.contractParams}
+      />
+    </>
+  )
 }
 
-export function ContractPageContent(
-  props: Parameters<typeof ContractPage>[0] & {
-    contract: Contract
-  }
-) {
+export function ContractPageContent(props: { contractParams: ContractParams }) {
+  const { contractParams } = props
   const {
     userPositionsByOutcome,
     comments,
     totalPositions,
-    pointsString,
     creatorTwitter,
     relatedContracts,
-  } = props
-  const contract = useContract(props.contract?.id) ?? props.contract
+  } = contractParams
+  const contract =
+    useContract(contractParams.contract?.id) ??
+    (contractParams.contract as Contract) ??
```

Comment:
Why cast as `Contract`? Isn't it already `Contract | null`?

Reply:
It's never null at that point since ContractPageContent doesn't show unless Contract is defined, I just don't know another way to redefine ContractParams

Reply:
Could do:
`export function ContractPageContent(props: { contractParams: ContractParams & { contract: Contract} }`